Source/WTF/wtf/Vector.h

```diff
@@ -56,48 +56,34 @@ DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(VectorBuffer);
 
 enum class NulloptBehavior : bool { Ignore, Abort };
 
-template <bool needsDestruction, typename T>
-struct VectorDestructor;
-
-template<typename T>
-struct VectorDestructor<false, T>
-{
-    static void destruct(T*, T*) { }
-};
-
 WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
 
 template<typename T>
-struct VectorDestructor<true, T>
+struct VectorTypeOperations
 {
-    static void destruct(T* begin, T* end) 
+    static void destruct(T* begin, T* end)
     {
-        for (T* cur = begin; cur != end; ++cur)
-            cur->~T();
+        if constexpr (!std::is_trivially_destructible_v<T>) {
+            for (T* cur = begin; cur != end; ++cur)
+                cur->~T();
+        }
     }
-};
-
-template <bool needsInitialization, bool canInitializeWithMemset, typename T>
-struct VectorInitializer;
 
-template<bool canInitializeWithMemset, typename T>
-struct VectorInitializer<false, canInitializeWithMemset, T>
-{
-    static void initializeIfNonPOD(T*, T*) { }
-
-    static void initialize(T* begin, T* end)
+    static void destruct(std::span<T> span)
     {
-        VectorInitializer<true, canInitializeWithMemset, T>::initialize(begin, end);
+        destruct(std::to_address(span.begin()), std::to_address(span.end()));
     }
-};
 
-template<typename T>
-struct VectorInitializer<true, false, T>
-{
     static void initializeIfNonPOD(T* begin, T* end)
     {
-        for (T* cur = begin; cur != end; ++cur)
-            new (NotNull, cur) T();
+        if constexpr (VectorTraits<T>::needsInitialization) {
+            if constexpr (VectorTraits<T>::canInitializeWithMemset)
+                memset(static_cast<void*>(begin), 0, reinterpret_cast<char*>(end) - reinterpret_cast<char*>(begin));
+            else {
+                for (T* cur = begin; cur != end; ++cur)
+                    new (NotNull, cur) T();
```

Comment:
In new code we would follow WebKit coding style and use a word rather than `cur`.

I think we can use `std::uninitialized_default_construct` or `std::ranges::uninitialized_default_construct` instead of writing a loop, although we will need to verify that it's still equally efficient.